### DISCUSSION TITLE
Chirped laser pulse in AMcylindrical geometry

### DISCUSSION THREAD

**Discussion by noreasonatall1** [SmileiPIC/Smilei#782] [Q&A] [answered]
Is there a straightforward way to define a chirped pulse in AMcylindrical geometry? Adding `chirp_profile` to `LaserGaussianAM` does not seem to work for me.

Thanks in advance and happy holidays to everyone!

**Comment by Z10Frank** [comment, marked as answer]
Hello,
the `LaserGaussianAM` profile in `src/Python/pyprofiles.py` is
```
def LaserGaussianAM( box_side="xmin", a0=1., omega=1., focus=None, waist=3.,
        polarization_phi=0., ellipticity=0., time_envelope=tconstant(), phase_offset=0.):
    from math import cos, sin, tan, atan, sqrt, exp
    if (len(focus)<1) or (len(focus)>2): 
        print("ERROR: focus should be a list of length 1")
        exit(1)
    elif (len(focus)==2):
        print("WARNING: deprecated focus in LaserGaussianAM should be a list of length 1")
    # Polarization and amplitude
    [dephasing, amplitudeY, amplitudeZ] = transformPolarization(polarization_phi, ellipticity)
    amplitudeY *= a0 * omega
    amplitudeZ *= a0 * omega
    # Space and phase envelopes
    Zr = omega * waist**2/2.
    w  = sqrt(1./(1.+(focus[0]/Zr)**2))
    invWaist2 = (w/waist)**2
    coeff = -omega * focus[0] * w**2 / (2.*Zr**2)
    def spatial(r):
        return w * exp( -invWaist2*(r)**2 )
    def phase(r):
        return coeff * (r)**2
    # Create Laser
    Laser(
        box_side        = box_side,
        omega          = omega,
        chirp_profile  = tconstant(),
        time_envelope  = time_envelope,
        space_envelope = [ lambda r:amplitudeZ*spatial(r), lambda r:amplitudeY*spatial(r) ],
        phase          = [ lambda r:phase(r)-phase_offset+dephasing, lambda r:phase(r)-phase_offset ],
        delay_phase    = [ 0., dephasing ]
    )
```

As you see the chirp profile is fixed to be constant. You can try defining a similar `Laser( ... )` block in the namelist with the desired chirp profile.

Since the definition of chirp profile may not correspond to other definitions that exist in the literature, alternatively you can specify a laser space-time profile from scratch in AMcylindrical geometry using the `space_time_profile_AM` https://smileipic.github.io/Smilei/Use/namelist.html#space_time_profile_AM

In this geometry you will need to specify the profile for each azimuthal mode, for all the transverse components at the window border. For a Gaussian beam, you will need for example to define the `Br`, `Bt` of mode `m=0` (which are identically 0) and the `Br`, `Bt` of mode `m=1`. In the input namelist for this tutorial https://smileipic.github.io/tutorials/advanced_vtk.html you will find an example of how to define a laser pulse using the `space_time_profile_AM` feature.

**Reply by noreasonatall1** [reply]
Sorry for the late update - specifying the `space_time_profile_AM` from scratch worked, although it took a bit of time. Thank you!

**Reply by Z10Frank** [reply]
You're welcome, thanks for the feedback!
